src/auth/digest/Config.cc

```diff
@@ -108,11 +108,14 @@ authDigestNonceEncode(digest_nonce_h * nonce)
     if (nonce->key)
         xfree(nonce->key);
 
-    nonce->key = xcalloc(base64_encode_len(sizeof(digest_nonce_data)), 1);
-    struct base64_encode_ctx ctx;
-    base64_encode_init(&ctx);
-    size_t blen = base64_encode_update(&ctx, reinterpret_cast<char*>(nonce->key), sizeof(digest_nonce_data), reinterpret_cast<const uint8_t*>(&(nonce->noncedata)));
-    blen += base64_encode_final(&ctx, reinterpret_cast<char*>(nonce->key)+blen);
+    SquidMD5_CTX Md5Ctx;
+    HASH H;
+    SquidMD5Init(&Md5Ctx);
+    SquidMD5Update(&Md5Ctx, reinterpret_cast<const uint8_t *>(&nonce->noncedata), sizeof(nonce->noncedata));
+    SquidMD5Final((unsigned char *) H, &Md5Ctx);
+
+    nonce->key = xcalloc(sizeof(HASHHEX), 1);
+    CvtHex(H, reinterpret_cast<char *>(nonce->key));
```

Comment:
The hash_link::key is a void* type. If you have to cast at all a static_cast should be sufficient.

Reply:
I changed it to `static_cast<char *>`. Not casting gave an error:
```
Config.cc: In function ‘void authDigestNonceEncode(digest_nonce_h*)’:
Config.cc:118:22: error: invalid conversion from ‘void*’ to ‘char*’ [-fpermissive]
     CvtHex(H, nonce->key);
               ~~~~~~~^~~
```